### PR TITLE
[New Feature] Consistent Dropout

### PR DESCRIPTION
**Pull request by N00bcak**
## Description

Introduces [Consistent Dropout (Hausknecht & Wagener, 2022)](https://arxiv.org/abs/2202.11818) to TorchRL.

Consists of the following changes: 
- `ConsistentDropout` PyTorch module 
- Companion `TensorDictModule`: `ConsistentDropoutModule`
- Tests for `ConsistentDropoutModule`

## Motivation and Context

Addresses a [feature request](https://discuss.pytorch.org/t/feature-request-consistent-dropout-implementation/183778).
Fleshes out a PR draft #1587.

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by vmoens** [comment]
I need to make a second pass over this but here's what I got running:

```python
from torchrl.modules import ConsistentDropoutModule, get_primers_from_module

from tensordict.nn import TensorDictModule as Mod, TensorDictSequential as Seq
import torch
from torchrl.envs import GymEnv
from tensordict import TensorDict

m = Seq(
    Mod(torch.nn.Linear(3, 4), in_keys=["observation"], out_keys=["intermediate"]),
    ConsistentDropoutModule(p=0.5, input_shape=(4,), in_keys="intermediate"),
    Mod(torch.nn.Linear(4, 1), in_keys=["intermediate"], out_keys=["action"]),
)
primer = get_primers_from_module(m)
env = GymEnv("Pendulum-v1").append_transform(primer)
r = env.reset()
env.rollout(32, m)["next", "mask_13266468624"]
```
With this you generate your mask during reset, such that it stays the same during execution.
